src/map/script.cpp

```diff
@@ -17235,6 +17235,96 @@ BUILDIN_FUNC(npcshopitem)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/*==========================================
+ *npcshopchange <npc name>,<itemid array>,<cost array>{,<market shop amount array>};
+ ******************************************/
+BUILDIN_FUNC(npcshopchange)
+{
+	const char* npcname = script_getstr(st, 2);
+	struct npc_data* nd = npc_name2id(npcname);
+	
+	struct script_data* item_dt;
+	struct script_data* cost_dt;
+	struct script_data* amount_dt = NULL;
+	const char* item_name;
+	const char* cost_name;
+	const char* amount_name = NULL;
+	int32 item_id, cost_id, amount_id = 0;
+	int32 item_idx, cost_idx, amount_idx = 0;
+	int i, item_hk, cost_hk, amount_hk = 0;
```

Comment:
Please, move all definitions right before where they're being used, also use t_itemid for item ids and nullptr instead of NULL

Reply:
Thanks for the review, refactored the code..
Saw some bug regarding clif marketshop. i'll create an issue and will look for it.